firmware/ci.c

```diff
@@ -263,6 +264,14 @@ static char *readstr(void)
 		if(readchar_nonblock()) {
 			c[0] = readchar();
 			c[1] = 0;
+
+			/* Check if we have anything to skip.  */
```

Comment:
I don't really like the reusage of `skip` variable here...

Maybe
```c
bool should_skip = (c[0] == skip);
skip = '\0';
if (should_skip)
  return NULL;
```

or

```c
if (c[0] == skip) {
  skip = '\0';
  return NULL;
} else {
  skip = '\0';
}
```

Reply:
I had it like the second case before.  I was just trying to be clever. I will revise.